grml-debootstrap

```diff
@@ -1507,96 +1497,96 @@ grub_install() {
     bailout 1
   fi
 
-  if [ -n "${ARM_EFI_TARGET}" ]; then
-    mkdir -p "${MNTPOINT}"/boot/efi
-    if ! mount "${ARM_EFI_TARGET}" "${MNTPOINT}"/boot/efi ; then
-      eerror "Error: Mounting ${ARM_EFI_TARGET} failed, can not continue."
-      bailout 1
-    fi
-  fi
-
   mount -t proc none "${MNTPOINT}"/proc
   mount -t sysfs none "${MNTPOINT}"/sys
   mount -t devtmpfs udev "${MNTPOINT}"/dev
   mount -t devpts devpts "${MNTPOINT}"/dev/pts
 
-  if [ -n "$ARM_EFI_TARGET" ]; then
-    einfo "Installing Grub as bootloader into EFI."
-
-    # shellcheck disable=SC2086
-    clean_chroot "$MNTPOINT" DEBIAN_FRONTEND=$DEBIAN_FRONTEND apt-get -y --no-install-recommends install $DPKG_OPTIONS grub-efi-arm64 grub-efi-arm64-signed
-
-    clean_chroot "${MNTPOINT}" grub-install --target=arm64-efi --efi-directory=/boot/efi --bootloader-id=debian --recheck --no-nvram --removable
   # Has chroot-script installed GRUB to MBR using grub-install (successfully), already?
   # chroot-script skips installation for unset ${GRUB}
-  elif [[ -z "${GRUB}" ]] || ! dd if="${GRUB}" bs=512 count=1 2>/dev/null | cat -v | grep -Fq GRUB; then
+  if [[ -z "${GRUB}" ]] || ! dd if="${GRUB}" bs=512 count=1 2>/dev/null | cat -v | grep -Fq GRUB; then
     einfo "Installing Grub as bootloader."
 
-    if ! clean_chroot "${MNTPOINT}" dpkg --list grub-pc 2>/dev/null | grep -q '^ii' ; then
-      echo "Notice: grub-pc package not present yet, installing it therefore."
-      # shellcheck disable=SC2086
-      clean_chroot "$MNTPOINT" DEBIAN_FRONTEND=$DEBIAN_FRONTEND apt-get -y --no-install-recommends install $DPKG_OPTIONS grub-pc
+    local grub_pc_package_name=''
+
+    # ARM64 uses EFI-only, and doesn't need a GRUB package. For i386, we only
+    # install the BIOS bootloader if we're not using EFI (since grub-pc and
+    # grub-efi-ia32 can't be co-installed). For amd64, we install grub-cloud
+    # which will pull in grub-pc-bin, so we don't need to explicitly install a
+    # BIOS bootloader for it when using EFI.
+    if [ -z "$VMEFI" ]; then
+      grub_pc_package_name='grub-pc'
     fi
 
-    mkdir -p "${MNTPOINT}/boot/grub"
-    if ! [ -d "${MNTPOINT}"/usr/lib/grub/i386-pc/ ] ; then
-      eerror "Error: grub not installed inside Virtual Machine. Can not install bootloader."
-      bailout 1
+    if [ -n "${grub_pc_package_name}" ]; then
+      if ! clean_chroot "${MNTPOINT}" dpkg --list "${grub_pc_package_name}" 2>/dev/null | grep -q '^ii' ; then
+        echo "Notice: ${grub_pc_package_name} package not present yet, installing it therefore."
+        clean_chroot "$MNTPOINT" DEBIAN_FRONTEND=$DEBIAN_FRONTEND apt-get -y --no-install-recommends install $DPKG_OPTIONS "${grub_pc_package_name}"
+      fi
     fi
-    cp -a "${MNTPOINT}"/usr/lib/grub/i386-pc "${MNTPOINT}/boot/grub/"
 
-    if [ -n "$VMEFI" ]; then
+    mkdir -p "${MNTPOINT}/boot/grub"
```

Comment:
This should be no longer relevant now that the `cp -a "${MNTPOINT}"/usr/lib/grub/i386-pc "${MNTPOINT}/boot/grub/"` is gone, nor?

Reply:
It does seem that way, I'll spin up some images with that bit omitted and see what happens.

Reply:
Removing this appears to work when building a Bookworm image with hybrid boot on a BIOS system.